lib/pavilion/series.py

```diff
@@ -36,12 +49,236 @@ def test_obj_from_id(pav_cfg, test_ids):
+            }
+
+            # resolve configs
+            # pylint: disable=W0212
+            configs_by_sched = run_cmd._get_test_configs(
+                pav_cfg=self.pav_cfg,
+                host=None,
+                test_files=[],
+                tests=[test_name],
+                modes=self.test_info[test_name]['modes'],
+                overrides=None,
+                sys_vars=sys_vars,
+                conditions=new_conditions
+            )
+
+            # configs -> test
+            # pylint: disable=W0212
+            tests_by_sched = run_cmd._configs_to_tests(
+                pav_cfg=self.pav_cfg,
+                configs_by_sched=configs_by_sched,
+                mb_tracker=mb_tracker,
+                build_only=False,
+                rebuild=False
+            )
+
+        except commands.CommandError as err:
+            # probably won't happen
+            err = codecs.decode(str(err), 'unicode-escape')
+            fprint(err, file=sys.stderr, flush=True)
+            self.finished.append(test_name)
+            return None
+
+        if tests_by_sched is None:
+            # probably won't happen but just in case
+            self.finished.append(test_name)
+            return errno.EINVAL
+
+        all_tests = sum(tests_by_sched.values(), [])
+        run_cmd.last_tests = list(all_tests)
+
+        if not all_tests:
+            # probably won't happen but just in case
+            fprint("You must specify at least one test.", file=sys.stderr)
+            self.test_info[test_name]['obj'] = run_cmd.last_tests
+            self.started.append(test_name)
+            return errno.EINVAL
+
+        # assign test to series and vice versa
+        self.series_obj.add_tests(all_tests)
+        run_cmd.last_series = self.series_obj
+
+        # make sure result parsers are ok
+        res = run_cmd.check_result_parsers(all_tests)
+        if res != 0:
+            # pylint: disable=W0212
+            run_cmd._complete_tests(all_tests)
+            self.test_info[test_name]['obj'] = run_cmd.last_tests
+            self.started.append(test_name)
+            return
+
+        # attempt to build
+        res = run_cmd.build_local(
+            tests=all_tests,
+            max_threads=self.pav_cfg.build_threads,
+            mb_tracker=mb_tracker,
+            build_verbosity=0
+        )
+        if res != 0:
+            # pylint: disable=W0212
+            run_cmd._complete_tests(all_tests)
+            self.test_info[test_name]['obj'] = run_cmd.last_tests
+            self.started.append(test_name)
+            return
+
+        run_cmd.run_tests(
+            pav_cfg=self.pav_cfg,
+            tests_by_sched=tests_by_sched,
+            series=self.series_obj,
+            wait=None,
+            report_status=False
+        )
+
+        self.test_info[test_name]['obj'] = run_cmd.last_tests
+        self.started.append(test_name)
+
+    # determines if test/s is/are done running
+    def is_done(self, test_name):
```

Comment:
You should periodically get the scheduler status of these tests when checking if they're done. They may have been cancelled by the scheduler, and died without setting their RUN_COMPLETE file. Asking for their status in this way will repair that problem automatically. You don't want to do it too often, maybe once ever 30 seconds or so.

Reply:
I added this in the `is_done` method.